### ISSUE TITLE
Do not fail fieldset[role=listbox] > caption for not having an accessible name

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2585]
**Expect**: The following should be excluded from the aria-input-field-name rule:

```html
<fieldset role="listbox">
  <legend>Favorite pizza</legend>
  <label><input type="radio" role="option">Cheese</label>
  <label><input type="radio" role="option">Cheese</label>
</fieldset>
```

**Actual**: The matcher of aria-input-field-name counts it, and fails it, even though legend will give this a name.

**Comment by padmavemulapati** [comment]
Verified for the above code -  `<fieldset role="listbox">`
  `<legend>Favorite pizza</legend>`
  `<label><input type="radio" role="option">Cheese</label>`
  `<label><input type="radio" role="option">Cheese</label>`
 `</fieldset> `
Its reproduced when tested with 4.0.0 version: `aria-input-field-name` rule is failing 
![image](https://user-images.githubusercontent.com/47409579/97797029-66e91200-1c3f-11eb-8b83-0562237b7c39.png)

Now validated with the latest 4.0.2v, not getting `aria-input-field-name` failure
![image](https://user-images.githubusercontent.com/47409579/97797020-4620bc80-1c3f-11eb-994b-1a566747d7a4.png)
